common/kokkos-sampler/kp_sampler_skip.cpp

```diff
@@ -1,323 +1,349 @@
+static endFunction endForCallee                = NULL;
+static endFunction endScanCallee               = NULL;
+static endFunction endReduceCallee             = NULL;
+
+void kokkosp_request_tool_settings(const uint32_t,
+                                   Kokkos_Tools_ToolSettings* settings) {
+  settings->requires_global_fencing = false;
+}
+
+// set of functions from Kokkos ToolProgrammingInterface (includes fence)
+Kokkos::Tools::Experimental::ToolProgrammingInterface tpi_funcs;
+
+uint32_t getDeviceID(uint32_t devid_in) {
+  int num_device_bits   = 7;
+  int num_instance_bits = 17;
+  return (~((uint32_t(-1)) << num_device_bits)) &
+         (devid_in >> num_instance_bits);
+}
+
+void invoke_ktools_fence(uint32_t devID) {
+  if (tpi_funcs.fence != nullptr) {
+    tpi_funcs.fence(devID);
+    if (tool_verbosity > 1) {
+      std::cout << "KokkosP: Sampler utility sucessfully invoked tool-induced "
+                   "fence on device "
+                << getDeviceID(devID) << ".\n";
+    }
+  } else {
+    std::cout << "KokkosP: FATAL: Kokkos Tools Programming Interface's "
+                 "tool-invoked Fence is NULL!\n";
+    exit(-1);
+  }
+}
+
+void kokkosp_provide_tool_programming_interface(
+    uint32_t num_funcs, Kokkos_Tools_ToolProgrammingInterface funcsFromTPI) {
+  if (!num_funcs) {
+    if (tool_verbosity > 0)
+      std::cout << "KokkosP: Note: Number of functions in Tools Programming "
+                   "Interface is 0!\n";
+  }
+  tpi_funcs = funcsFromTPI;
+}
+
+void kokkosp_init_library(const int loadSeq, const uint64_t interfaceVer,
+                          const uint32_t devInfoCount, void* deviceInfo) {
+  const char* tool_verbose_str   = getenv("KOKKOS_TOOLS_SAMPLER_VERBOSE");
+  const char* tool_globFence_str = getenv("KOKKOS_TOOLS_GLOBALFENCES");
+  if (NULL != tool_verbose_str) {
+    tool_verbosity = atoi(tool_verbose_str);
+  } else {
+    tool_verbosity = 0;
+  }
+  if (NULL != tool_globFence_str) {
+    tool_globFence = atoi(tool_globFence_str);
+  } else {
+    tool_globFence = 0;
+  }
+
+  char* profileLibrary = getenv("KOKKOS_TOOLS_LIBS");
+  if (NULL == profileLibrary) {
+    printf(
+        "Checking KOKKOS_PROFILE_LIBRARY. WARNING: This is a depreciated "
+        "variable. Please use KOKKOS_TOOLS_LIBS\n");
+    profileLibrary = getenv("KOKKOS_PROFILE_LIBRARY");
+    if (NULL == profileLibrary) {
+      std::cout << "KokkosP: FATAL: No library to call in " << profileLibrary
+                << "!\n";
+      exit(-1);
+    }
+  }
+
+  char* envBuffer = (char*)malloc(sizeof(char) * (strlen(profileLibrary) + 1));
+  strcpy(envBuffer, profileLibrary);
+
+  char* nextLibrary = strtok(envBuffer, ";");
+
+  for (int i = 0; i < loadSeq; i++) {
+    nextLibrary = strtok(NULL, ";");
+  }
+
+  nextLibrary = strtok(NULL, ";");
+
+  if (NULL == nextLibrary) {
+    std::cout << "KokkosP: FATAL: No child library of sampler utility library "
+                 "to call in "
+              << profileLibrary << "!\n";

```

Comment:
If this is a fatal error, we should abort (and print to std::cerr) instead, right?

Reply:
I have taken out the setting of global fences as you requested. 

(Also, I was getting confused on if you were saying to get rid of global fences in sampling altogether).